src/idl_gen_general.cpp

```diff
@@ -203,8 +202,9 @@ class GeneralGenerator : public BaseGenerator {
       const std::string &classcode, bool needs_includes) {
     if (!classcode.length()) return true;
 
-    std::string code;
-    code = code + "// " + FlatBuffersGeneratedWarning();
+    std::string code("// <auto-generated>\n");
+    code = code + "// " + FlatBuffersGeneratedWarning() + "\n// </auto-generated>\n";
```

Comment:
Sorry to be pedantic, but:
- This file is used for Java code gen too. Unless Java has the same convention, it probably shouldn't go in there.
- Why is the closing tag `</auto-generated>` directly after the warning? From the discussion elsewhere I was expecting the closing tag to either be at the end of the file (enclosing all generated code), or omitted entirely.